mypy/errors.py

```diff
@@ -122,6 +122,58 @@ def __init__(self,
                    Optional[ErrorCode]]
 
 
+class ErrorWatcher:
+    """Context manager that can be used to keep track of new errors recorded
+    around a given operation.
+
+    Errors maintain a stack of such watchers. The handler is called starting
+    at the top of the stack, and is propagated down the stack unless filtered
+    out by one of the ErrorWatcher instances.
+    """
+    def __init__(self, errors: 'Errors', *,
+                 filter_errors: Union[bool, Callable[[str, ErrorInfo], bool]] = False,
+                 save_filtered_errors: bool = False):
+        self.errors = errors
+        self._has_new_errors = False
+        self._filter = filter_errors
+        self._filtered: Optional[List[ErrorInfo]] = [] if save_filtered_errors else None
+
+    def __enter__(self) -> 'ErrorWatcher':
+        self.errors._watchers.append(self)
+        return self
+
+    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> Literal[False]:
```

Comment:
```suggestion
    def __exit__(self, exc_type: object, exc_val: object, exc_tb: object) -> Literal[False]:
```
Avoid Any if possible